desc/profiles.py

```diff
@@ -77,6 +229,300 @@ def __repr__(self):
             + " (name={}, grid={})".format(self.name, self.grid)
         )
 
+    def __mul__(self, x):
+        """Multiply this profile by another or a constant."""
+        if np.isscalar(x):
+            return ScaledProfile(x, self)
+        elif isinstance(x, Profile):
+            return ProductProfile(self, x)
+        else:
+            raise NotImplementedError()
+
+    def __rmul__(self, x):
+        """Multiply this profile by another or a constant."""
+        return self.__mul__(x)
+
+    def __add__(self, x):
+        """Add this profile with another."""
+        if isinstance(x, Profile):
+            return SumProfile(self, x)
+        else:
+            raise NotImplementedError()
+
+    def __neg__(self):
+        """Invert the sign of this profile."""
+        return ScaledProfile(-1, self)
+
+    def __sub__(self, x):
+        """Subtract another profile from this one."""
+        return self.__add__(-x)
+
+
+class ScaledProfile(Profile):
+    """Profile times a constant value.
+
+    f_1(x) = a*f(x)
+
+    Parameters
+    ----------
+    profile : Profile
+        Base profile to scale.
+    scale : float
+        Scale factor.
+
+    """
+
+    _io_attrs_ = Profile._io_attrs_ + ["_profile", "_scale"]
+
+    def __init__(self, scale, profile, **kwargs):
+        assert isinstance(
+            profile, Profile
+        ), "profile in a ScaledProfile must be a Profile or subclass, got {}.".format(
+            str(profile)
+        )
+        assert np.isscalar(scale), "scale must be a scalar."
+
+        self._profile = profile.copy()
+        self._scale = scale
+
+        kwargs.setdefault("name", profile.name)
+        kwargs.setdefault("grid", profile.grid)
+        super().__init__(**kwargs)
+
+    @property
+    def grid(self):
+        """Default grid for computation."""
+        return self._grid
+
+    @grid.setter
+    def grid(self, new):
+        Profile.grid.fset(self, new)
+        self._profile.grid = new
+
+    @property
+    def params(self):
+        return (self._scale, self._profile.params)
+
+    @params.setter
+    def params(self, x):
+        if isinstance(x, tuple) and len(x) == 2:
```

Comment:
very minor but maybe consider `isinstance(lst, (list, tuple))` on the `params` setters since people use these interchangeably

Reply:
good catch